File: src/main/java/me/nobaboy/nobaaddons/util/CooldownManager.java
package me.nobaboy.nobaaddons.util;

public class CooldownManager {
	private long markedAt = 0;
	private long cooldownDuration = 3000L;

	public void startCooldown() {
		this.markedAt = System.currentTimeMillis();
	}

	public void startCooldown(long duration) {
		this.startCooldown();
		this.cooldownDuration = duration;
	}

	public boolean isOnCooldown() {
		long expiresAt = markedAt + cooldownDuration;
		return expiresAt > System.currentTimeMillis();
	}
}

File: src/main/kotlin/me/nobaboy/nobaaddons/util/CooldownManager.kt
package me.nobaboy.nobaaddons.util

open class CooldownManager {
    private var markedAt: Long = 0
    private var cooldownDuration: Long = 3000L

    fun startCooldown() {
        this.markedAt = System.currentTimeMillis()
    }

    fun startCooldown(duration: Long) {
        this.startCooldown()
        this.cooldownDuration = duration
    }

    fun isOnCooldown(): Boolean {
        val expiresAt = markedAt + cooldownDuration
        return expiresAt > System.currentTimeMillis()
    }
}